src/psyclone/tests/psyGen_test.py

```diff
@@ -3258,74 +3258,82 @@ def test_symbol_scalar_array():
-    sym = Symbol('a', 'real', shape=[], scope='global_argument',
-                 is_input=False, is_output=False)
-    sym.is_output = True
-    assert sym.is_output is True
 
-    with pytest.raises(TypeError) as error:
-        sym.is_output = 3
-    assert "Symbol 'is_output' attribute must be a boolean." in \
-           str(error.value)
+def test_symbol_interface_access():
+    ''' Tests for the SymbolInterface.access setter. '''
+    symbol = Symbol("some_var", "real",
+                    interface=Symbol.FortranGlobal(module_use="my_mod"))
+    symbol.interface.access = Symbol.Access.READ
+    assert symbol.interface.access == Symbol.Access.READ
+    # Force the error by supplying a string instead of a SymbolAccess type.
+    with pytest.raises(TypeError) as err:
+        symbol.interface.access = "read"
+    assert "must be a 'Symbol.Access' but got " in str(err)
 
-    sym = Symbol('a', 'real', [], 'local')
-    with pytest.raises(ValueError) as error:
-        sym.is_output = True
-    assert ("Symbol with 'local' scope can not have 'is_output' attribute"
-            " set to True.") in str(error)
 
+def test_symbol_argument_str():
+    ''' Check the __str__ method of the Symbol.Argument class. '''
+    # A Symbol.Argument represents a routine argument by default.
+    interface = Symbol.Argument()
+    assert str(interface) == "Argument(pass-by-value=False)"
 
-def test_symbol_can_be_printed():
-    '''Test that a Symbol instance can always be printed. (i.e. is
-    initialised fully)'''
-    sym1 = Symbol("s1", "integer")
-    assert "s1: <integer, local, Scalar>" in str(sym1)
 
-    sym2 = Symbol("s2", "real", [None, 2, sym1])
-    assert "s2: <real, local, Array['Unknown bound', 2, s1]>" in str(sym2)
+def test_fortranglobal_str():
+    ''' Test the __str__ method of Symbol.FortranGlobal. '''
+    # If it's not an argument then we have nothing else to say about it (since
+    # other options are language specific and are implemented in sub-classes).
+    interface = Symbol.FortranGlobal("my_mod")
+    assert str(interface) == "FortranModule(my_mod)"
 
-    sym2._shape.append('invalid')
-    with pytest.raises(InternalError) as error:
-        _ = str(sym2)
-    assert ("Symbol shape list elements can only be 'Symbol', 'integer' or "
-            "'None', but found") in str(error.value)
+
+def test_fortranglobal_modname():
+    ''' Test the FortranGlobal.module_name setter error conditions. '''
+    with pytest.raises(ValueError) as err:
+        _ = Symbol.FortranGlobal("")
+    assert "module_name must be one or more characters long" in str(err)
+    with pytest.raises(TypeError) as err:
+        _ = Symbol.FortranGlobal(1)
+    assert "module_name must be a str but got" in str(err)
 
     sym3 = Symbol("s3", "integer", constant_value=12)
-    assert "s3: <integer, local, Scalar, constant_value=12>" in str(sym3)
+    assert "s3: <integer, Scalar, local, constant_value=12>" in str(sym3)
```

Comment:
This 2 lines have also jumped from the `test_symbol_can_be_printed()`

Reply:
Oh yes. Apologies for missing that.